ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemAdd.java

```diff
@@ -375,6 +380,30 @@ protected void execute(DeploymentProcessorTarget processorTarget) {
             EJB3SubsystemDefaultEntityBeanOptimisticLockingWriteHandler.INSTANCE.updateOptimisticLocking(context, model);
         }
 
+        if (model.hasDefined(SERVER_INTERCEPTORS)) {
+            final List<ServerInterceptorMetaData> serverInterceptors = new ArrayList<>();
+
+            final ModelNode serverInterceptorsNode = model.get(SERVER_INTERCEPTORS);
+            for(final ModelNode serverInterceptor: serverInterceptorsNode.asList()){
+                serverInterceptors.add(new ServerInterceptorMetaData(serverInterceptor.get("module").asString(), serverInterceptor.get("class").asString()));
+            }
+
+            context.addStep(new AbstractDeploymentChainStep() {
+                protected void execute(DeploymentProcessorTarget processorTarget) {
+                    processorTarget.addDeploymentProcessor(EJB3Extension.SUBSYSTEM_NAME, Phase.DEPENDENCIES, Phase.DEPENDENCIES_EJB_SERVER_INTERCEPTORS,
+                            new ServerInterceptorsDependenciesDeploymentUnitProcessor(serverInterceptors));
+                }
+            }, OperationContext.Stage.RUNTIME);
+
+            final ServerInterceptorCache serverInterceptorCache = new ServerInterceptorCache(serverInterceptors);
+            context.addStep(new AbstractDeploymentChainStep() {
+                protected void execute(DeploymentProcessorTarget processorTarget) {
+                    processorTarget.addDeploymentProcessor(EJB3Extension.SUBSYSTEM_NAME, Phase.POST_MODULE, Phase.POST_MODULE_EJB_SERVER_INTERCEPTORS,
+                            new ServerInterceptorsBindingsProcessor(serverInterceptorCache));
+                }
+            }, OperationContext.Stage.RUNTIME);
+        }
```

Comment:
Please don't add new steps for these DUPs. These DUPs can be added as part of the existing AbstractDeploymentChainStep at L259.

Reply:
fixed